src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx

```diff
@@ -306,19 +305,16 @@ export function OpenSavedQueryFlyout({
               fill
               onClick={() => {
                 if (selectedQuery) {
-                  if (
-                    // Template queries are not associated with data sources. Apply data source from current query
-                    selectedQuery.attributes.isTemplate
-                  ) {
-                    const updatedQuery: Query = {
-                      ...queryStringManager?.getQuery(),
-                      query: selectedQuery.attributes.query.query,
-                      language: selectedQuery.attributes.query.language,
-                    };
-                    queryStringManager.setQuery(updatedQuery);
-                  } else {
-                    onQueryOpen(selectedQuery);
-                  }
+                  onQueryOpen({
+                    ...selectedQuery,
+                    attributes: {
+                      ...selectedQuery.attributes,
+                      query: {
+                        ...selectedQuery.attributes.query,
+                        dataset: queryStringManager.getQuery().dataset,
+                      },
+                    },
+                  });
```

Comment:
is this all to make sure we're not unsetting the current dataset when a query is selected?

Reply:
Yes